DataFormats/interface/KInfo.h

```diff
@@ -174,7 +174,7 @@ struct KGenEventInfo : public KEventInfo
 	inline float getLheWeight(size_t index, bool failOnError = true) const
 	{
 		if(failOnError)
-			assert(lheWeight[index] < float(-998.9) && lheWeight[index] > float(999.1) ); // the user tried to access something that has not been properly filled during the skim
+			assert(!(lheWeight[index] < float(-998.9) || lheWeight[index] > float(999.1)) ); // the user tried to access something that has not been properly filled during the skim
```

Comment:
Is the cut value abs(999)? Than you could use <=-999 and >=999 instead of these floats

Reply:
This might actually stongly depend on the MC sample. So my suggestion would be to drop it, or change the default behaviour to `failOnError = false`. There are only a few cases where it is resonable to cut on the eventWeight.

Reply:
I change the default behaviour